### PR TITLE
Increase jacoco's min coverage to 70%

### PR DESCRIPTION
**Pull request by thdinizm**
With the new tests in this PR, the common package is at 90% code and 75% branches.

phone-numbers package is already above that threshold

### CODE REVIEW

File: sdk/communication/azure-communication-common/src/test/java/com/azure/communication/common/implementation/HmacAuthenticationPolicyTests.java [hunk 1]
```diff
@@ -245,7 +245,6 @@ public void httpRequestTest() throws MalformedURLException {
 
         HttpRequest request = new HttpRequest(HttpMethod.GET, new URL("http://localhost?id=b93a5ef4-f622-44d8-a80b-ff983122554e"));
         StepVerifier.create(pipeline.send(request))
-            .expectError(RuntimeException.class);
+            .verifyError(RuntimeException.class);
```
Comment (paolamvhz) [inline sdk/communication/azure-communication-common/src/test/java/com/azure/communication/common/implementation/HmacAuthenticationPolicyTests.java:248]: Why this has been changed?

File: sdk/communication/azure-communication-common/src/test/java/com/azure/communication/common/UnknownIdentifierTests.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import static org.junit.jupiter.api.Assertions.*;
+
+public class UnknownIdentifierTests {
+
+    final String id = "some id";
+
+    @Test
+    public void constructWithNullOrEmptyIdShouldThrow() {
+        assertThrows(IllegalArgumentException.class, () ->
+            new UnknownIdentifier(null), "Should throw on null id");
```
Comment (paolamvhz) [inline sdk/communication/azure-communication-common/src/test/java/com/azure/communication/common/UnknownIdentifierTests.java:16]: Can we change this for "The initialization parameter [id] cannot be null" ?

File: sdk/communication/azure-communication-common/src/test/java/com/azure/communication/common/UnknownIdentifierTests.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+        assertThrows(IllegalArgumentException.class, () ->
+            new UnknownIdentifier(""), "Should throw on empty id");
```
Comment (paolamvhz) [inline sdk/communication/azure-communication-common/src/test/java/com/azure/communication/common/UnknownIdentifierTests.java:19]: Can we change this for "The initialization parameter [id] cannot be empty"?

File: sdk/communication/azure-communication-common/src/test/java/com/azure/communication/common/implementation/HmacAuthenticationPolicyTests.java (hunk 1 above)
Comment (thdinizm) [reply sdk/communication/azure-communication-common/src/test/java/com/azure/communication/common/implementation/HmacAuthenticationPolicyTests.java:248]: I'm not 100% sure on the difference between `expectError` vs `verifyError`, but this test is supposed to fail at the `http vs https` check. If you debug the code, it was failing somewhere else before.

**Comment by minnieliu** [comment]
> /azp run java - communication - tests

This is not the command that we use anymore to run the java live tests. 
We have divided the pipeline into new ones: https://dev.azure.com/azure-sdk/internal/_build?pipelineNameFilter=java%20-%20azure-

File: sdk/communication/azure-communication-phonenumbers/pom.xml [hunk 4]
```diff
@@ -49,8 +49,8 @@
   <properties>
     <src.dir>src/main</src.dir>
     <test.dir>src/test</test.dir>
-    <jacoco.min.linecoverage>0.68</jacoco.min.linecoverage>
-    <jacoco.min.branchcoverage>0.66</jacoco.min.branchcoverage>
+    <jacoco.min.linecoverage>0.70</jacoco.min.linecoverage>
```
Comment (minnieliu) [inline sdk/communication/azure-communication-phonenumbers/pom.xml:52]: We skip some tests related to Purchase and Release in our live test pipeline but not locally. Updating the coverage here may cause failures in the live test pipeline as coverage may not be met

File: sdk/communication/azure-communication-common/pom.xml [hunk 5]
```diff
@@ -38,8 +38,8 @@
   <properties>
     <src.dir>src/main</src.dir>
     <test.dir>src/test</test.dir>
-    <jacoco.min.linecoverage>0.69</jacoco.min.linecoverage>
-    <jacoco.min.branchcoverage>0.52</jacoco.min.branchcoverage>
+    <jacoco.min.linecoverage>0.70</jacoco.min.linecoverage>
```
Comment (minnieliu) [inline sdk/communication/azure-communication-common/pom.xml:41]: Nice! 👍 This is great to see

File: sdk/communication/azure-communication-common/src/test/java/com/azure/communication/common/CommunicationUserIdentifierTests.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    public void constructWithNullOrEmptyIdShouldThrow() {
+        assertThrows(IllegalArgumentException.class, () ->
+            new CommunicationUserIdentifier(null), "The initialization parameter [id] cannot be null");
+
+        assertThrows(IllegalArgumentException.class, () ->
+            new CommunicationUserIdentifier(""), "The initialization parameter [id] cannot be empty");
+    }
+
+    @Test
+    public void compareEqualUnknownIdentifiers() {
```
Comment (minnieliu) [inline sdk/communication/azure-communication-common/src/test/java/com/azure/communication/common/CommunicationUserIdentifierTests.java:23]: This is for "UserIdentifier" right? Not "UnknownIdentifiers"?

File: sdk/communication/azure-communication-common/src/test/java/com/azure/communication/common/CommunicationUserIdentifierTests.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        CommunicationUserIdentifier identifier1 =  new CommunicationUserIdentifier(id);
+        CommunicationUserIdentifier identifier2 =  new CommunicationUserIdentifier(id);
+
+        assertTrue(identifier1.equals(identifier1));
+        assertTrue(identifier1.equals(identifier2));
+    }
+
+    @Test
+    public void compareWithNonUnknownIdentifier() {
```
Comment (minnieliu) [inline sdk/communication/azure-communication-common/src/test/java/com/azure/communication/common/CommunicationUserIdentifierTests.java:32]: Same here

**Review by minnieliu — APPROVED** [review]

**Pull request merged**